runtimes/runtimes/agent.ts

```
@@ -48,7 +55,12 @@ export const newAgent = (): Agent => {
             }
 
             if (!tool.validate(input, token)) {
-                throw new Error(`Input for tool ${toolName} is invalid`)
+                const errorDetails =
```

I see you're setting the validationError outside of this scope at the tool level to reuse here, but it might be cleaner if we can just return the error from `tool.validate()`?

---

tool.validate() will not return any error, the error is from the validator

---

yes but we control what tool.validate returns

---

fixed